pkg/converter/tool/feature.go

```diff
@@ -70,44 +89,49 @@ func detectVersion(msg []byte) string {
 }
 
 // DetectFeatures returns supported feature list from required feature list.
-func DetectFeatures(builder string, required Features) Features {
+func DetectFeatures(builder string, required Features) (Features, error) {
 	currentVersionDetectOnce.Do(func() {
-		if required.Contains(FeatureTar2Rafs) && disableTar2Rafs {
-			logrus.Warnf("the feature '%s' is disabled by env '%s'", FeatureTar2Rafs, envNydusDisableTar2Rafs)
-		}
+		requiredFeatures = required
+		detectedFeatures = Features{}
 
 		cmd := exec.CommandContext(context.Background(), builder, "--version")
 		output, err := cmd.Output()
 		if err != nil {
 			return
 		}
 
-		currentVersion = detectVersion(output)
-	})
-
-	if currentVersion == "" {
-		return Features{}
-	}
-
-	detectedFeatures := Features{}
-	for _, feature := range required {
-		requiredVersion := featureMap[feature]
-		if requiredVersion == "" {
-			detectedFeatures = append(detectedFeatures, feature)
-			continue
+		currentVersion := detectVersion(output)
+		if currentVersion == "" {
+			return
 		}
 
-		// The feature is supported by current version
-		supported := semver.Compare(requiredVersion, "v"+currentVersion) <= 0
-		if supported {
-			// It is an experimental feature, so we still provide an env
-			// variable to allow users to disable it.
-			if feature == FeatureTar2Rafs && disableTar2Rafs {
+		for _, feature := range required {
+			requiredVersion := featureMap[feature]
+			if requiredVersion == "" {
+				detectedFeatures = append(detectedFeatures, feature)
 				continue
 			}
-			detectedFeatures = append(detectedFeatures, feature)
+
+			// The feature is supported by current version
+			supported := semver.Compare(requiredVersion, "v"+currentVersion) <= 0
+			if supported {
+				// It is an experimental feature, so we still provide an env
+				// variable to allow users to disable it.
+				if feature == FeatureTar2Rafs && disableTar2Rafs {
+					logrus.Warnf("the feature '%s' is disabled by env '%s'", FeatureTar2Rafs, envNydusDisableTar2Rafs)
+					continue
+				}
+				detectedFeatures = append(detectedFeatures, feature)
+			} else {
+				logrus.Warnf("the feature '%s' requirement not satisfied: nydus-image >= %s, current: v%s, omitted", feature, featureMap[feature], currentVersion)
```

Comment:
`the feature '%s' is ignored, it requires nydus-image >= v%s (current: v%s)`

Reply:
Please double check this warning.